### ISSUE TITLE
[DataGrid] date and datetime types in darkmode

### ISSUE THREAD

**Issue by daveteu** [mui-org/material-ui-x#2379]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

In DataGrid, when using date and dateTime types, the Date and DateTimePicker components are not respecting `palette.darkmode`

![image](https://user-images.githubusercontent.com/18649392/129556120-d8edddd6-4947-470d-8d77-84ea7dde96b6.png)

## Expected Behavior 🤔

Picker should respect theme colors

This behavior can also be observed on Mui Documentations - https://material-ui.com/components/data-grid/editing/#controlled-editing

## Your Environment 🌎

<details>
  <summary>System:
 System

</summary>
  
```
 System:
    OS: macOS 11.4
  Binaries:
    Node: 16.5.0 - ~/.nvm/versions/node/v16.5.0/bin/node
    Yarn: Not Found
    npm: 7.19.1 - ~/.nvm/versions/node/v16.5.0/bin/npm
  Browsers:
    Chrome: 92.0.4515.131
    Edge: Not Found
    Firefox: Not Found
    Safari: 14.1.1
  npmPackages:
    @material-ui/core: ^5.0.0-beta.2 => 5.0.0-beta.2 
    @material-ui/data-grid: ^4.0.0-alpha.37 => 4.0.0-alpha.37 
    @material-ui/icons: ^5.0.0-beta.1 => 5.0.0-beta.1 
    @material-ui/lab: ^5.0.0-alpha.41 => 5.0.0-alpha.41 
    @material-ui/private-theming:  5.0.0-beta.2 
    @material-ui/styled-engine-sc: ^5.0.0-beta.1 => 5.0.0-beta.1 
    @material-ui/styles: ^5.0.0-beta.2 => 5.0.0-beta.2 
    @material-ui/system:  5.0.0-beta.2 
    @material-ui/types:  6.0.1 
    @material-ui/unstyled:  5.0.0-alpha.41 
    @material-ui/utils:  5.0.0-beta.1 
    @types/react: ^17.0.0 => 17.0.11 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    styled-components: ^5.3.0 => 5.3.0 
    typescript: ^4.1.3 => 4.3.2 

```
</details>

**Comment by oliviertassinari** [comment]
<img width="718" alt="Capture d’écran 2021-08-16 à 16 29 23" src="https://user-images.githubusercontent.com/3165635/129580085-8cb71025-e299-41b5-8277-9acbc1963e0f.png">

Duplicate of https://github.com/mui-org/material-ui/issues/25016

**Comment by oliviertassinari** [comment]
I think that we can solve this issue the same way we will #1805: with https://github.com/mui-org/material-ui/issues/25016, and using the CssBaseline component.

**Comment by m4theushw** [comment]
Fixed by mui-org/material-ui#25016.

You need to use `<CssBaseline enableColorScheme />` in your theme. See https://mui.com/components/css-baseline/#page
